cmd/storage-server/server.go

```diff
@@ -69,6 +90,36 @@ func (s *storageServer) PutFile(ctx context.Context, req *spb.PutFileRequest) (*
 	return &spb.PutFileResponse{}, nil
 }
 
+func (s *storageServer) PutFileStream(stream spb.StorageService_PutFileStreamServer) error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	in, err := stream.Recv()
+	if err != nil {
+		return err
+	}
+	// First message determines the key, it will be assumed that all subsequent
+	// file keys are the same
```

Comment:
Is this always guaranteed? Is it possible for two streams to be interleaved during processing?

Reply:
No, streams will never be interleaved